src/formatDate.js

```diff
@@ -8,7 +8,50 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  let year;
+  let month;
+  let day;
+  const formattedDate = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      year = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'YY') {
+      year = dateParts[i] < 30 ? `20${dateParts[i]}` : `19${dateParts[i]}`;
```

Comment:
The condition `dateParts[i] < 30` works because JavaScript performs type coercion, but it's comparing a string to a number, which can be misleading. It's better to explicitly convert `dateParts[i]` to a number before the comparison for clarity and to avoid any potential type coercion issues.